packages/amazon/src/serverGroup/configure/serverGroupCommandBuilder.service.js

```diff
@@ -359,49 +378,127 @@ angular
               const shouldAutoEnableIPv6 =
                 asgSettings && asgSettings.enableIPv6 && asgSettings.setIPv6InTest && isTestEnv;
 
+              const secGroups = launchTemplateData.networkInterfaces
+                ? (launchTemplateData.networkInterfaces.find((ni) => ni.deviceIndex === 0) || {}).groups
+                : launchTemplateData.securityGroups.length && launchTemplateData.securityGroups;
+
```

Comment:
The assignment of `command.securityGroups` was previously happening at the end of the function with repeated conditionals like `if serverGroup.launchConfig / if serverGroup.launchTemplate` [here](https://github.com/spinnaker/deck/blob/master/packages/amazon/src/serverGroup/configure/serverGroupCommandBuilder.service.js#L390-L399). Instead of adding a new conditional `if serverGroup.mixedInstancesPolicy` for the 2nd time in the same function, I consolidated the assignment of `command.securityGroups` with other assignments for each case (launchConfig, launchTemplate, mixedInstancesPolicy). 

I have tested a number of use cases with the new code with launchConfig, launchTemplate and mixedInstancesPolicy - see testing details [here](https://github.com/spinnaker/spinnaker/issues/5989#issuecomment-920303624)

I can just revert these changes if these are a cause of concern.

Reply:
The changes are tested for various use cases like launch config, launch template, mixed instances policy, and a number of edge cases (manually + new unit / integ testes). See details [here](https://github.com/spinnaker/spinnaker/issues/5989#issuecomment-920303624). 

This change was made to consolidate the conditional assignments to command's properties. I can revert this to previous version if this is a cause of concern.